cli/util/uploader.go

```diff
@@ -0,0 +1,446 @@
+package util
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/cli/cmdargs"
+	"github.com/nspcc-dev/neo-go/cli/options"
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/services/oracle/neofs"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/nspcc-dev/neofs-sdk-go/checksum"
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/pool"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/version"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	// Number of objects to search in a batch for finding max block in container.
+	searchBatchSize = 10000
+	// Control the number of concurrent searches.
+	maxParallelSearches = 40
+	// Size of object ID.
+	oidSize = sha256.Size
+
+	// Number of workers to fetch and upload blocks concurrently.
+	numWorkers = 100
+
+	// Maximum number of retries.
+	maxRetries = 3
+	// Initial backoff duration.
+	initialBackoff = 500 * time.Millisecond
+	// Backoff multiplier.
+	backoffFactor = 2
+	// Maximum backoff duration.
+	maxBackoff = 10 * time.Second
```

Comment:
Not implemented as suggested. Move it to a separate block of constants, add comment to this block that these constants related to retry mechanism.